File: TaskManager.cs
using System;

public interface IPrinter
{
    void Print();
}

public interface IScanner
{
    void Scan();
}

public class Printer : IPrinter
{
    public void Print()
    {
        Console.WriteLine("Printing document...");
    }
}

public class Scanner : IScanner
{
    public void Scan()
    {
        Console.WriteLine("Scanning document...");
    }
}

public class PrintScanner : IPrinter, IScanner
{
    
    private readonly IPrinter _printer;
    private readonly IScanner _scanner;

    public PrintScanner(IPrinter printer, IScanner scanner)
    {
        _printer = printer;
        _scanner = scanner;
    }

    public void Print()
    {
        Console.WriteLine("PrintScanner: Printing document...");
        _printer.Print();
    }

    public void Scan()
    {
        Console.WriteLine("PrintScanner: Scanning document...");
        _scanner.Scan();
    }
}

public class TaskManager
{
    public void PrintTask(int taskId, IPrinter printer)
    {
        Console.WriteLine($"Executing Print Task: {taskId}");
        printer.Print();
    }

    public void ScanTask(int taskId, IScanner scanner)
    {
        Console.WriteLine($"Executing Scan Task: {taskId}");
        scanner.Scan();
    }
}

public class Program
{
    public static void Main()
    {
        var printer = new Printer();
        var scanner = new Scanner();
        var printScanner = new PrintScanner(printer,scanner);

        var scheduler = new TaskManager();

        scheduler.PrintTask(101, printer);
        scheduler.ScanTask(102, scanner);

        scheduler.PrintTask(103, printScanner);
        scheduler.ScanTask(104, printScanner);
    }
}

File: TaskManager.py
from abc import ABC, abstractmethod


class IPrinter(ABC):
    @abstractmethod
    def print(self):
        pass


class IScanner(ABC):
    @abstractmethod
    def scan(self):
        pass


class Printer(IPrinter):
    def print(self):
        print("Printing document...")


class Scanner(IScanner):
    def scan(self):
        print("Scanning document...")


class PrintScanner(IPrinter, IScanner):
    def print(self):
        print("PrintScanner: Printing document...")

    def scan(self):
        print("PrintScanner: Scanning document...")


class TaskManager:
    def print_task(self, task_id, printer: IPrinter):
        print(f"Executing Print Task: {task_id}")
        printer.print()

    def scan_task(self, task_id, scanner: IScanner):
        print(f"Executing Scan Task: {task_id}")
        scanner.scan()


printer = Printer()
scanner = Scanner()
printScanner = PrintScanner()

scheduler = TaskManager()

scheduler.print_task(101, printer)
scheduler.scan_task(102, scanner)

scheduler.print_task(103, printScanner)
scheduler.scan_task(104, printScanner)